File: api-v2/app/com/foreignlanguagereader/api/dto/v1/document/WordDTO.scala
package com.foreignlanguagereader.api.dto.v1.document

import com.foreignlanguagereader.api.dto.v1.definition.DefinitionDTO
import play.api.libs.json.{Format, Json, Reads}

case class WordDTO(token: String,
                   tag: String,
                   lemma: String,
                   definitions: Option[Seq[DefinitionDTO]])

object WordDTO {
  implicit val format: Format[WordDTO] = Json.format
  implicit val reads: Reads[WordDTO] =
    Json.reads[WordDTO]
  implicit val readsSeq: Reads[Seq[WordDTO]] =
    Reads.seq(reads)
}

File: dto/src/main/java/com/foreignlanguagereader/dto/v1/word/WordDTO.java
package com.foreignlanguagereader.dto.v1.word;

import com.foreignlanguagereader.dto.v1.definition.DefinitionDTO;

import java.util.List;

public class WordDTO {
    private final String token;
    private final String processedToken;
    private final String tag;
    private final String lemma;

    public WordDTO(String token, String processedToken, String tag, String lemma) {
        this.token = token;
        this.processedToken = processedToken;
        this.tag = tag;
        this.lemma = lemma;
    }

    public String getTag() {
        return tag;
    }

    public String getLemma() {
        return lemma;
    }

    public String getToken() {
        return token;
    }

    public String getProcessedToken() {
        return processedToken;
    }
}
